spec/dummy/spec/lib/send_grid_mailer/definition_spec.rb

```diff
@@ -30,6 +30,30 @@
     end
   end
 
+  describe "#dynamic_template_data" do
+    before do
+      @substitution = subject.dynamic_template_data({ key: 'value' })
+    end
+
+    it "creates substitution with valid data" do
+      expect(@substitution).to eq({ key: 'value' })
+    end
+
+    it "adds substitution to personalization object" do
+      expect(personalization.dynamic_template_data.size).to eq(1)
+    end
+
+    it "add dynamic_data to data hash" do
+      subject.dynamic_template_data({ other_key: 'other_value' })
```

Comment:
Name your test subject if you need to reference it explicitly.<br>Redundant curly braces around a hash parameter.